Clockwise/VirtualClock.cs

```diff
@@ -8,10 +8,9 @@ namespace Clockwise
 {
     public class VirtualClock : IClock, IDisposable
     {
+        private readonly List<(DateTimeOffset, Action<VirtualClock>)> schedule = new List<(DateTimeOffset, Action<VirtualClock>)>();
         private static readonly Logger logger = Logger<VirtualClock>.Log;
 
-        private readonly SortedList<DateTimeOffset, Action<VirtualClock>> schedule = new SortedList<DateTimeOffset, Action<VirtualClock>>();
```

Comment:
Sorted lists are great, but pulling off of the "0th" element causes an O(n) operation on every deque.  The List should be sorted backwards so that popping off the next scheduled item is O(1).

Also, the SortedList class is actually built out of two lists, one for the key, one for the value.  This is not great for cache locality on dequeue.